### PR TITLE
Update non transaction retry bbe 

### PR DESCRIPTION
**Pull request by pcnfernando**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by mohanvive — APPROVED** [review]

File: examples/non-transaction-retry/non_transaction_retry.bal [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import ballerina/io;
+import ballerina/lang.'error as errors;
 
 public function main() returns error? {
     // The retry statement provides a general-purpose retry
```
Comment (praneesha) [inline examples/non-transaction-retry/non_transaction_retry.bal:5]:
```suggestion
    // The retry statement provides a general-purpose retry.
```

File: examples/non-transaction-retry/non_transaction_retry.bal [hunk 2]
```diff
@@ -11,7 +12,19 @@ public function main() returns error? {
         io:println("Attempting execution...");
         // Calls a function, which simulates an error scenario to 
         // trigger the retry operation.
-        return doWork();
+        check doWork();
+    }
+
+    int i = 0;
+
+    // You can pass a retry manager class as a type parameter
```
Comment (praneesha) [inline examples/non-transaction-retry/non_transaction_retry.bal:20]:
```suggestion
    // You can pass a retry manager class as a type parameter.
```

File: examples/non-transaction-retry/non_transaction_retry.bal [hunk 3]
```diff
@@ -21,8 +34,26 @@ int count = 0;
...
+        // Return a retriable error so that
+        // the default retry manager retries.
+        return errors:Retriable("Execution Error");
     } else {
-        io:println("Work done.");
+        io:println("Work is done.");
     }
 }
+
+// Retry manager class with arbitrarily logic.
```
Comment (praneesha) [inline examples/non-transaction-retry/non_transaction_retry.bal:45]:
```suggestion
// Retry the manager class with an arbitrarily logic.
```

File: examples/non-transaction-retry/non_transaction_retry.out [hunk 4]
```diff
@@ -3,4 +3,7 @@
 ballerina run non_transaction_retry.bal
 Attempting execution...
 Attempting execution...
-Work done.
+Work is done.
```
Comment (praneesha) [inline examples/non-transaction-retry/non_transaction_retry.out:6]:
```suggestion
Work completed.
```

File: examples/non-transaction-retry/non_transaction_retry.out [hunk 5]
```diff
@@ -3,4 +3,7 @@
...
+Attempting execution...
+Attempting execution...
+Work is done.
```
Comment (praneesha) [inline examples/non-transaction-retry/non_transaction_retry.out:9]:
```suggestion
Work completed..
```
Why is this line repeated? Saying "Work is done/completed is a bit ambiguous. Can we specify more details about what was done?

File: examples/non-transaction-retry/tests/non_transaction_retry_test.bal [hunk 6]
```diff
@@ -16,5 +16,8 @@ function testFunc() {
     error? output = main();
     test:assertEquals(outputs[0], "Attempting execution...");
     test:assertEquals(outputs[1], "Attempting execution...");
-    test:assertEquals(outputs[2], "Work done.");
+    test:assertEquals(outputs[2], "Work is done.");
```
Comment (praneesha) [inline examples/non-transaction-retry/tests/non_transaction_retry_test.bal:19]:
```suggestion
    test:assertEquals(outputs[2], "Work completed.");
```
Same comments as above.

File: examples/non-transaction-retry/tests/non_transaction_retry_test.bal [hunk 7]
```diff
@@ -16,5 +16,8 @@ function testFunc() {
...
+    test:assertEquals(outputs[3], "Attempting execution...");
+    test:assertEquals(outputs[4], "Attempting execution...");
+    test:assertEquals(outputs[5], "Work is done.");
```
Comment (praneesha) [inline examples/non-transaction-retry/tests/non_transaction_retry_test.bal:22]:
```suggestion
    test:assertEquals(outputs[5], "Work completed.");
```
Same comments as above.

File: examples/non-transaction-retry/non_transaction_retry.bal (hunk 3 above)
Comment (pcnfernando) [reply examples/non-transaction-retry/non_transaction_retry.bal:45]: this is to say, pass a retry manager class which can have any arbitrary logic

**Review by praneesha — APPROVED** [review]

**Pull request merged**